### PR TITLE
values.yaml - helm chart docs edits for 0.49.x

### PR DESCRIPTION
**Pull request by david-yu**
Changes proposed in this PR:
- Missing changes from https://github.com/hashicorp/consul/pull/15452 that were not back ported to values.yaml also see https://github.com/hashicorp/consul/pull/15849

How I've tested this PR:

How I expect reviewers to test this PR:

Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/values.yaml [hunk 1]
```diff
@@ -776,7 +776,7 @@ server:
   # storage classes, the PersistentVolumeClaims would need to be manually created.
   # A `null` value will use the Kubernetes cluster's default StorageClass. If a default
   # StorageClass does not exist, you will need to create one.
-  # Refer to the [Read/Write Tuning](https://www.consul.io/docs/install/performance#read-write-tuning) 
+  # Refer to the [Read/Write Tuning](/docs/install/performance#read-write-tuning)
```
Comment (nathancoleman) [inline charts/consul/values.yaml:779]: Will this link work?

Comment (david-yu) [reply charts/consul/values.yaml:779]: It should generate something like 

```
 Refer to the [Read/Write Tuning](/consul/docs/install/performance#read-write-tuning)
     section of the Server Performance Requirements documentation for considerations
```

which does work from mdx and is the convention we use in our docs.

Comment (nathancoleman) [reply charts/consul/values.yaml:779]: that makes sense, but I was thinking people aren't looking at a `values.yaml` file through the docs site which is why it probably had the hostname included

Comment (david-yu) [reply charts/consul/values.yaml:779]: What's interesting is that this pattern seems to be already be used in https://github.com/hashicorp/consul-k8s/blob/main/charts/consul/values.yaml#L2321 and https://github.com/hashicorp/consul-k8s/blob/64e8011be5c1eaf7b32ff6d697d6f9fc71cb3699/charts/consul/values.yaml#L303. 

I need to defer to the Consul docs team as to what might be the best recommendation here.

Comment (nathancoleman) [reply charts/consul/values.yaml:779]: FWIW this isn't something I have a strong opinion on. Was asking more out of curiosity.

**Review by nathancoleman — APPROVED** [review]

File: charts/consul/values.yaml (hunk 1 above)
Comment (im2nguyen) [reply charts/consul/values.yaml:779]: can we use the absolute links? this updates it to the latest absolute link

```suggestion
  # Refer to the [Read/Write Tuning](https://developer.hashicorp.com/consul/docs/install/performance#read-write-tuning) 
```

**Review by im2nguyen — APPROVED** [review]

**Pull request merged**